multiapps-controller-core/src/main/java/org/cloudfoundry/multiapps/controller/core/validators/parameters/HostValidator.java

```diff
@@ -1,15 +1,39 @@
 package org.cloudfoundry.multiapps.controller.core.validators.parameters;
 
+import java.util.Collections;
 import java.util.Map;
+import java.util.Set;
 
-import org.apache.commons.lang3.StringUtils;
+import org.cloudfoundry.multiapps.common.util.MapUtil;
 import org.cloudfoundry.multiapps.controller.core.Messages;
 import org.cloudfoundry.multiapps.controller.core.model.SupportedParameters;
+import org.cloudfoundry.multiapps.controller.core.util.NameUtil;
 
 public class HostValidator extends RoutePartValidator {
 
+    private final String namespace;
+    private final boolean applyNamespaceGlobal;
+
+    public HostValidator() {
+        this.namespace = null;
+        this.applyNamespaceGlobal = false;
+    }
+
     public HostValidator(String namespace, boolean applyNamespaceGlobal) {
-        super(namespace, applyNamespaceGlobal);
+        this.namespace = namespace;
+        this.applyNamespaceGlobal = applyNamespaceGlobal;
+    }
+
+    @Override
+    public boolean isValid(Object routePart, final Map<String, Object> context) {
+        boolean validRoutePart = super.isValid(routePart, context);
+        if (!validRoutePart) {
+            return false;
+        }
+        if (shouldApplyNamespace(context)) {
+            return ((String) routePart).startsWith(namespace);
```

Comment:
Why? StringUtils.startsWith require again to cast Object to String. The only difference which I see is that if `namespace` is null, it will not throw NPE but this scenario shouldn't be possible to occur.

Reply:
I agree, it cannot be null, but it just seems a bit more error-prone without a null check but ok